packages/idempotency/src/middleware/makeHandlerIdempotent.ts

```diff
@@ -72,6 +72,15 @@ const makeHandlerIdempotent = (
     request: MiddyLikeRequest,
     retryNo = 0
   ): Promise<unknown | void> => {
+    if (
```

Comment:
To avoid having to call the static method at each step, as far as I know, we have two options that are of varying degrees of idiomatic in Middy-land:

1. define a variable in the outer scope (~L50) like `let shouldSkipIdempotency = false`, call the static method in the `before` & store the result in the variable `shouldSkipIdempotency = IdempotencyHandler.shouldSkipIdempotency(...)`, and then use this variable to make the decision in the `after` & `error`.
2. Do the same, but store it instead [Middy's internal storage](https://middy.js.org/docs/writing-middlewares/internal-storage). Note that the docs talk about promises, but you can store any kind of values in it. Later stages (`after` or `error`) can then access the store from the request object (i.e. `request.internal. shouldSkipIdempotency`) - this is [what we do here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/3312019f11ffe0067eef18fe9d66aa8dcb58b0f9/packages/commons/src/middleware/cleanupMiddlewares.ts#L68).

Either way, knowing how the jmespath module is implemented, the savings in terms of ops are marginal since the results for the same pair of object & JMESPath expression are cached.

Reply:
I pick option 1 for now. We need to revisit middleware implementation in a separate issue to refcator IdempotencyHandler, so we don't have two implementations to deal with.

Reply:
Created an issue to track this: #1508